lib/SlackBot.js

```diff
@@ -225,113 +225,253 @@ function Slackbot(configuration) {
+            //     // Receive messages and trigger events from the Events API
+            //     res.send('');
+            //     return handleEventsAPI(payload, bot);
+            // } else if (payload.callback_id) {
+            //     // is this an interactive message callback?
+            //     res.send('');
+            //     return handleInteractiveMessage(payload, bot);
+            // } else if (payload.command) {
+            //     // this is a slash command
+            //     return handleSlashCommand(payload, bot);
+            // } else if (payload.trigger_word) {
+            //     return handleOutgoingWebhook(payload, bot);
+            // }
         });
     };
 
+
+
+    // Send a 200 response back to Slack to acknowledge the message.
+    slack_botkit.middleware.ingest.use(function sendResponse(bot, message, res, next) {
+
+        if (res && res.statusCode) {
+            // this is an http response
+            // always send a 200
+            res.status(200);
+
+            // conditionally send a response back to Slack to acknowledge the message.
+            // we do NOT want to respond to incoming webhooks or slash commands
+            // as the response can be used by developers to actually deliver a reply
+            if (!message.command && !message.trigger_word) {
+                res.send('');
+            }
+        }
+        next();
+
+    });
+
+    /* do delivery confirmations for RTM messages */
+    slack_botkit.middleware.ingest.use(function requireDelivery(bot, message, res, next) {
+        if (message.ok != undefined) {
+            // this is a confirmation of something we sent.
+            if (slack_botkit.config.require_delivery) {
+                // loop through all active conversations this bot is having
+                // and mark messages in conversations as delivered = true
+                for (var t = 0; t < slack_botkit.tasks.length; t++) {
+                    if (slack_botkit.tasks[t].isActive()) {
+                        for (var c = 0; c < slack_botkit.tasks[t].convos.length; c++) {
+                            for (var s = 0; s < slack_botkit.tasks[t].convos[c].sent.length; s++) {
+                                if (slack_botkit.tasks[t].convos[c].sent[s].api_response && slack_botkit.tasks[t].convos[c].sent[s].api_response.id == message.reply_to) {
+                                    slack_botkit.tasks[t].convos[c].sent[s].delivered = true;
+                                    slack_botkit.tasks[t].convos[c].sent[s].api_response.ts = message.ts;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        next();
+    });
+
+
+
+    slack_botkit.middleware.categorize.use(function(bot, message, next) {
+
+        var mentionSyntax = '<@' + bot.identity.id + '(\\|' + bot.identity.name.replace('.', '\\.') + ')?>';
```

Comment:
`bot.identity.name` can sometimes be undefined, can we be a bit more defensive here? `bot.identity.name.replace` will throw an error if name is undefined.